### PR TITLE
 API routes to connect to Google calendar

### PR DESCRIPTION
**Pull request by yesyash**
### Changes 
- Made `/google/connect` & `/google/callback` to be able to authorize and get tokens from Google Calendar. 

### Note
This is step 1 of #40 and currently does not have the ability to create AccessTokens post-authorization this will be done in step 3 of the ticket.

### CODE REVIEW

File: config/custom-environment-variables.ts [hunk 1]
```diff
@@ -23,6 +23,13 @@ module.exports = {
     logLevel: "LOGS_LOG_LEVEL",
   },
 
+  calendarAccessToken: {
```
Comment (ankurnarkhede) [inline config/custom-environment-variables.ts:26]: Cant we use the existing credentials here from `providers.googleOauth20`?

File: controllers/calendar.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from "express";
+import { google } from "googleapis";
+import config from "config";
+
+const gcalClientId = config.get("calendarAccessToken.gcal.clientId");
+const gcalClientSecret = config.get("calendarAccessToken.gcal.clientSecret");
+const calApiUrl = config.get("services.calendarApi.baseUrl");
+
+const oauth2Client = new google.auth.OAuth2(
```
Comment (ankurnarkhede) [inline controllers/calendar.ts:9]: Can we define this inside `googleConnectHandler()` as this isnt used anywhere else?

File: controllers/calendar.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const googleCallbackHandler = async (
+  _: Request,
+  res: Response
+): Promise<void> => {
+  const redirectUrl = `${String(
+    config.get("services.rCalUi.baseUrl")
+  )}/onboarding`;
+
+  res.redirect(redirectUrl);
```
Comment (ankurnarkhede) [inline controllers/calendar.ts:34]: Lets add consistent return in all controller functions...

File: controllers/calendar.ts (hunk 2 above)
Comment (yesyash) [reply controllers/calendar.ts:9]: This will be needed in phase 3 of the ticket #40

**Review by harshith-venkatesh — APPROVED** [review]
Good work @yesyash , apart from Ankur's comment rest all looks good.
Could we also track why we are getting warning and how we could resolve that also please

File: controllers/calendar.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const url = oauth2Client.generateAuthUrl({
+    access_type: "offline",
+    scope: scopes,
+  });
+
+  return res.redirect(url);
+};
+
+const googleCallbackHandler = (_: Request, res: Response): void => {
+  const redirectUrl = `${String(
```
Comment (ankurnarkhede) [inline controllers/calendar.ts:27]: Arent we storing the tokens here?

Comment (ankurnarkhede) [reply controllers/calendar.ts:27]: And the calendar IDs?

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: controllers/calendar.ts (hunk 4 above)
Comment (ankurnarkhede) [reply controllers/calendar.ts:27]: Seems that is supposed to be a part of the issue https://github.com/Real-Dev-Squad/calendar-API/issues/40.
Merging this.

**Pull request merged**